frontend/src/components/NewProposalCommentForm.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect, useMemo } from "react";
+import toast from "react-hot-toast";
+import { useAccount, useContractWrite } from "@starknet-react/core";
+import { submitCommentApi } from "../api/apiService";
+import { CONTRACT_ADDR, formatAddress } from "../lib/config";
+import { shortString } from "starknet";
+export default function NewcommentCommentForm({
+	setIsModalOpen,
+	propId,
+}: {
+	setIsModalOpen: React.Dispatch<React.SetStateAction<boolean>>;
+	propId: string;
+}) {
+	const { isConnected, address } = useAccount();
+	const [comment, setComment] = React.useState<string>("");
+	const [ipfsHash, setIpfsHash] = React.useState<string>("");
+	const [isLoading, setIsLoading] = React.useState<boolean>(false);
+
+	const calls = useMemo(() => {
```

Comment:
Should useMemo really be used to trigger side effects such as prompting the signature of a tx? Wouldn't useEffect be more appropriate?